src/components/common/Autocomplete/CurrencyAutocomplete/CurrencyAutocomplete.tsx

```diff
@@ -0,0 +1,52 @@
+import { Autocomplete, AutocompleteProps, TextField } from '@mui/material';
+import { useApiConstants } from 'src/components/Constants/UseApiConstants';
+
+interface CurrencyAutocompleteProps
+  extends Partial<AutocompleteProps<any, boolean, boolean, boolean>> {
```

Comment:
The first generic type is the type of the options, which are strings here.

You probably have to update options to `options={currencies.map((cur) => cur.code).filter((code): code is string => typeof code === 'string')}` to make TypeScript happy since the currency codes can be `undefined`.

```suggestion
  extends Partial<AutocompleteProps<string, boolean, boolean, boolean>> {
```